components/BottomAppBar/src/private/MDCBottomAppBarLayer.m

```diff
@@ -59,77 +70,85 @@ - (CGPathRef)pathFromRect:(CGRect)rect
                 shouldCut:(BOOL)shouldCut {
   UIBezierPath *bottomBarPath = [UIBezierPath bezierPath];
 
-  CGFloat arcRadius =
-      CGRectGetHeight(floatingButton.bounds) / 2 + kMDCBottomAppBarFloatingButtonRadiusOffset;
-  CGFloat navigationBarYOffset = CGRectGetMinY(navigationBarFrame);
-  CGFloat halfAngle = acosf((float)((navigationBarYOffset - floatingButton.center.y) / arcRadius));
-  CGFloat startAngle = (float)M_PI / 2 + halfAngle;
-  CGFloat endAngle = (float)M_PI / 2 - halfAngle;
+  CGFloat yOffset = CGRectGetMinY(navigationBarFrame);
 
   CGFloat width = CGRectGetWidth(rect);
   CGFloat height = CGRectGetHeight(rect);
+  CGFloat arcRadius =
+      CGRectGetHeight(floatingButton.bounds) / 2 + kMDCBottomAppBarFloatingButtonRadiusOffset;
+
+  CGPoint arcCenter1 =
+      CGPointMake(floatingButton.center.x - CGRectGetWidth(floatingButton.bounds) / 2 + arcRadius -
+                      kMDCBottomAppBarFloatingButtonRadiusOffset,
+                  floatingButton.center.y);
+  CGPoint arcCenter2 =
+      CGPointMake(floatingButton.center.x + CGRectGetWidth(floatingButton.bounds) / 2 - arcRadius +
+                      kMDCBottomAppBarFloatingButtonRadiusOffset,
+                  floatingButton.center.y);
 
   if (shouldCut) {
-    [self drawWithPathToCut:bottomBarPath
```

Comment:
test is failing because it can't find this API;

error: -[BottomAppBarTests testPathToAndFromEqualNumberOfPoints] : failed: caught "NSInvalidArgumentException", "-[MDCBottomAppBarLayer drawWithPathToCut:yOffset:width:height:arcCenter:arcRadius:startAngle:endAngle:]: unrecognized selector sent to instance 0x7fdb1c309fb0"
(
	0   CoreFoundation                      0x000000010b5eab0b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010b04f141 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010b65a134 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x000000010b571840 ___forwarding___ + 1024
	4   CoreFoundation                      0x000000010b5713b8 _CF_forwarding_prep_0 + 120
	5   unit_tests_IPHONE_7_PLUS_IN_10_3    0x000000011a63b444 -[BottomAppBarTests testPathToAndFromEqualNumberOfPoints] + 324
	6   CoreFoundation                      0x000000010b572c6c __invoking___ + 140
	7   CoreFoundation                      0x000000010b572b40 -[NSInvocation invoke] + 320
	8   XCTest                              0x000000010af22e30 __24-[XCTestCase invokeTest]_block_invoke + 591
	9   XCTest                              0x000000010af6c17e -[XCUITestContext performInScope:] + 183
	10  XCTest                              0x000000010af22bd6 -[XCTestCase invokeTest] + 141
	11  XCTest                              0x000000010af23b97 __26-[XCTestCase performTest:]_block_invoke.369 + 42
	12  XCTest                              0x000000010af70f25 +[XCTContext runInContextForTestCase:block:] + 163
	13  XCTest                              0x000000010af23533 -[XCTestCase performTest:] + 608
	14  XCTest                              0x000000010af1f539 __27-[XCTestSuite performTest:]_block_invoke + 363
	15  XCTest                              0x000000010af1eea0 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 26
	16  XCTest                              0x000000010af1f09d -[XCTestSuite performTest:] + 239
	17  XCTest                              0x000000010af1f539 __27-[XCTestSuite performTest:]_block_invoke + 363
	18  XCTest                              0x000000010af1eea0 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 26
	19  XCTest                              0x000000010af1f09d -[XCTestSuite performTest:] + 239
	20  XCTest                              0x000000010af1f539 __27-[XCTestSuite performTest:]_block_invoke + 363
	21  XCTest                              0x000000010af1eea0 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 26
	22  XCTest                              0x000000010af1f09d -[XCTestSuite performTest:] + 239
	23  XCTest                              0x000000010af7864f __44-[XCTTestRunSession runTestsAndReturnError:]_block_invoke + 40
	24  XCTest                              0x000000010af3271a -[XCTestObservationCenter _observeTestExecutionForBlock:] + 475
	25  XCTest                              0x000000010af784ee -[XCTTestRunSession runTestsAndReturnError:] + 281
	26  XCTest                              0x000000010af0eaf1 -[XCTestDriver runTestsAndReturnError:] + 314
	27  XCTest                              0x000000010af70190 _XCTestMain + 619
	28  xctest                              0x000000010a909591 xctest + 9617
	29  libdyld.dylib                       0x00000001125dc65d start + 1
)

Reply:
Just uploaded a fix for the tests.